tests/python/pants_test/backend/python/tasks/test_ctypes_integration.py

```diff
@@ -42,12 +43,24 @@ def test_ctypes_run(self):
 
     self.assertEqual('x=3, f(x)=17\n', pants_run.stdout_data)
 
-  def test_ctypes_binary(self):
+  def test_ctypes_binary_creation(self):
+    """Create a python_binary() with all native toolchain variants, and test the result."""
+    # TODO: this pattern could be made more ergonomic for `enum()`, along with exhaustiveness
+    # checking.
+    for variant in ToolchainVariant.allowed_values:
+      self._assert_ctypes_binary(variant)
+
+  def _assert_ctypes_binary(self, toolchain_variant):
+    # TODO: figure out a way to check that when we select 'gnu' as the `toolchain_variant`, we use
```

Comment:
Maybe you could use `ldebug` and pattern match against what exe we use?

Reply:
I was thinking of reading the output, hadn't thought of using `ldebug`. When I tried to set `'level': 'debug'` in the pants run config, it failed with an error relating to run tracking -- however, I realized that knowing the compiler and linker's filenames isn't that intrusive of a message and might be really useful as an info log. So the current commit checks the compiler and linker filenames are in the output from the info logs that were added.